scipy/io/wavfile.py

```
@@ -728,7 +765,10 @@ def read(filename, mmap=False):
         if not hasattr(filename, 'read'):
             fid.close()
         else:
-            fid.seek(0)
```

I am frankly mystified as to why this final seek operation was originally here. Most other libraries will not seek to the beginning of a file if you hand them an open file to read - e.g. if you use `pd.read_csv()` to read a file, the file position after the function call will be where it stopped reading.

I would suggest removing `fid.seek(0)`, and not seeking even if the file supports it.

Thoughts, @j-bowhay ?

---

It was added in 47b9012, @nils-werner do you happen to remember?

---

If I remember correctly it was there before, I just proposed the `if` clause. Maybe follow the `git blame` to the changes before?

I don't think it needs to be the at all tbh.

---

`seek(0)` doesn't seem to appear prior to 47b9012, maybe it had something to do with StringIO compatibility?

Removing it makes sense to me, but I worry it could break user code that depends on that behavior.

---

Ah right, IIRC that was to fill the buffer with RIFF data, and then rewind it so that the buffer can immediately be used by something that wants to read a WAV file (which contains RIFF data).

The use case was the `Audio` tag in Jupiter, where you'd want to export an array to a WAV file-like buffer, and then send it over to the frontend.

---

Ah ok, that makes sense. I've added a comment.

---

Commit dd3568a34f6a19b67755b93934e910c55f9dbde2 looks like a nice compromise: it does not swallow exceptions if the underlying stream claims to support seeking, but actually can't. This was my primary concern. It also doesn't change functionality in case applications are relying on the `seek(0)` behavior.